### ISSUE TITLE
[ENHANCEMENT] Add AIC bias correction for small sample sizes with AICc

### ISSUE THREAD

**Issue by martinvonk** [pastas/pastas#725]
**Describe the proposed enhancement**
I am running into the case quite often where the AIC between two models still prefers the model with more parameters. A fit difference in the R$^2$ of 1% still generally prefers the highest fit model.

I want to play around with the AIC by adding the AICc which also corrects for the number of observations. Generally we have a lot of observations, but I think it is worth it to implement it. 

**Expected behavior**
```python
def aic(obs: pd.Series, sim: pd.Series, n_param: int) -> float:
   ...

def aicc(obs: pd.Series, sim: pd.Series, n_param: int) -> float:
   K = n_param
   n = len(obs)
   c_term = (2*K * (K+1)) / (n-K-1)
   return aic(obs=obs, sim=sim, n_param=n_param) + c_term
```

note AICc converges to the AIC if `n -> infinity`.

**Comment by martinvonk** [comment]
related to #303

**Comment by raoulcollenteur** [comment]
I think this issue can be closed after PR #727 was merged right?
